src/workspace-data/data-table/wds/WDSContent.test.ts

```diff
@@ -0,0 +1,114 @@
+import { RecordTypeSchema } from 'src/libs/ajax/data-table-providers/WdsDataTableProvider';
+import { renderWithAppContexts as render } from 'src/testing/test-utils';
+import { defaultAzureWorkspace } from 'src/testing/workspace-fixtures';
+
+import WDSContent, { WDSContentProps } from './WDSContent';
+
+jest.mock('src/libs/error', () => ({
+  ...jest.requireActual('src/libs/error'),
+  reportError: jest.fn(),
+}));
+
+jest.mock('src/libs/notifications', () => ({
+  notify: jest.fn(),
+}));
+
+interface SetupOptions {
+  props: WDSContentProps;
+  entities: Entity[];
+}
+
+const trafficLightSchema: RecordTypeSchema = {
+  name: 'stop-light-signal',
+  count: 1,
+  attributes: [
+    { name: 'id', datatype: 'number' },
+    { name: 'color', datatype: 'string' },
+    { name: 'shape', datatype: 'string' },
+    { name: 'flashing', datatype: 'boolean' },
+  ],
+  primaryKey: 'id',
+};
+const defaultProps: WDSContentProps = {
+  workspace: {
+    ...defaultAzureWorkspace,
+    workspace: {
+      ...defaultAzureWorkspace.workspace,
+      // attributes are required to avoid an error while destructuring from 'workspace-column-defaults'
+      attributes: {},
+    },
+  },
+  recordType: trafficLightSchema.name,
+  wdsSchema: [trafficLightSchema],
+  editable: true,
+  dataProvider: {} as DataTableProvider,
+};
+
+const defaultSetupOptions: SetupOptions = {
+  props: defaultProps,
+  entities: [
+    {
+      name: 'stop',
+      entityType: 'stop-light-signal',
+      attributes: { color: 'red', shape: 'octagon', flashing: false },
+    },
+    {
+      name: 'slow',
+      entityType: 'stop-light-signal',
+      attributes: { color: 'yellow', shape: 'triangle', flashing: false },
+    },
+    {
+      name: 'go',
+      entityType: 'stop-light-signal',
+      attributes: { color: 'green', shape: 'circle', flashing: false },
+    },
+    {
+      name: 'wtf',
+      entityType: 'stop-light-signal',
+      attributes: { color: 'purple', shape: 'trapezoid', flashing: true },
+    },
+  ],
+};
+describe('WDSContent', () => {
+  const setup = ({ props, entities }: SetupOptions = defaultSetupOptions) => {
+    const getPageResponse: DeepPartial<EntityQueryResponse> = {
+      results: entities,
+      resultMetadata: {
+        unfilteredCount: entities.length,
+        filteredCount: entities.length,
+        filteredPageCount: entities.length / 100 + 1,
+      },
+    };
+
+    const dataProvider: DeepPartial<DataTableProvider> = {
+      getPage: jest.fn().mockResolvedValue(getPageResponse),
+      features: {
+        supportsCapabilities: true,
+        supportsAttributeDeleting: true,
+      },
+    };
+
+    return { props: { ...props, dataProvider: dataProvider as DataTableProvider } };
+  };
+
+  it('renders without crashing', async () => {
+    // Arrange
+    const { props } = setup({
+      ...defaultSetupOptions,
+    });
+
+    // Act
+    await act(() => {
+      render(h(WDSContent, props));
+    });
+
+    await act(() => screen.debug());
```

Comment:
I haven't yet been able to get any of the fixture to actually render into the HTML, current state:
<img width="771" alt="image" src="https://github.com/DataBiosphere/terra-ui/assets/111856/c7b04464-18eb-488b-b0e2-92f09aaf5b4f">

Reply:
I suspect this is due to the react-virtualized AutoSizer in DataTable. It renders the table based on available space. In tests' JSDOM environment, it thinks there is no space available and renders no rows.

Here's an example of how to mock AutoSizer
https://github.com/DataBiosphere/terra-ui/blob/6cd172e42681d0f27a8c705d66b1eafa00f57d96/src/components/file-browser/FilesTable.test.ts#L10-L15

Reply:
Thanks I'll try this!